# HG changeset patch
# User Kai T. Ohlhus <k.ohlhus@gmail.com>
# Date 1471359462 -7200
#      Tue Aug 16 16:57:42 2016 +0200
# Node ID 3732bee6951067f749aa955e2b163e69e6b40a0a
# Parent  2cd1afd0f12f3259a4bee14dfcbfd44d6ec10ad1
publish.m: math formatting prevention for more than nine formulas.

diff --git a/scripts/general/publish.m b/scripts/general/publish.m
--- a/scripts/general/publish.m
+++ b/scripts/general/publish.m
@@ -824,17 +824,17 @@ function str = format_text (str, formatt
     ## Bold
     str = regexprep (str, '\*([^*$_|]*)\*', formatter ("bold", "$1"));
     ## Italic
     str = regexprep (str, '_([^_$|*]*)_', formatter ("italic", "$1"));
     ## Monospaced
     str = regexprep (str, '\|([^|$_*]*)\|', formatter ("monospaced", "$1"));
   until (strcmp (str, oldstr))
   ## Restore inline "$" and block "$$" LaTeX math
-  for i = 1:length(math_cstr)
+  for i = length(math_cstr):-1:1
     str = strrep (str, ["PUBLISHMATH", num2str(i)], math_cstr{i});
   endfor
   ## Replace special symbols
   str = strrep (str, "(TM)", formatter("TM"));
   str = strrep (str, "(R)", formatter("R"));
 endfunction
 
 
